demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +98,144 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
 
     return blobs;
 }
+
+void EncoderDecoderCNN::check_net_names(const OutputsDataMap &output_info_encoder,
+                                        const OutputsDataMap &output_info_decoder,
+                                        const InputsDataMap &input_info_decoder
+                                        ) {
+    std::string name_not_exist = "";
+    if (output_info_encoder.find(out_enc_hidden_name_) == output_info_encoder.end())
+        name_not_exist = out_enc_hidden_name_;
+    if (output_info_encoder.find(features_name_) == output_info_encoder.end())
+        name_not_exist = features_name_;
+    if (input_info_decoder.find(in_dec_hidden_name_) == input_info_decoder.end())
+        name_not_exist = in_dec_hidden_name_;
+    if (input_info_decoder.find(features_name_) == input_info_decoder.end())
+        name_not_exist = features_name_;
+    if (input_info_decoder.find(in_dec_symbol_name_) == input_info_decoder.end())
+        name_not_exist = in_dec_symbol_name_;
+    if (output_info_decoder.find(out_dec_hidden_name_) == output_info_decoder.end())
+        name_not_exist = out_dec_hidden_name_;
+    if (output_info_decoder.find(out_dec_symbol_name_) == output_info_decoder.end())
+        name_not_exist = out_dec_symbol_name_;
+    if (name_not_exist != "")
+        throw NameNotExist(name_not_exist);
+ }
+
+
+EncoderDecoderCNN::EncoderDecoderCNN(const std::string &model_path,
+                                     Core & ie, const std::string & deviceName,
+                                     const std::string out_enc_hidden_name,
+                                     const std::string out_dec_hidden_name,
+                                     const std::string in_dec_hidden_name,
+                                     const std::string features_name,
+                                     const std::string in_dec_symbol_name,
+                                     const std::string out_dec_symbol_name,
+                                     const std::string logits_name,
+                                     unsigned end_token,
```

Comment:
`kAlphabet.find()` returns `size_t`. `std::distance` returns signed. Use one of them, not the third type. I suggest `size_t`.

Reply:
Changed to size_t